File: 2021/1/task1.go
package main

import (
    "fmt"
    "io/ioutil"
    s "strings"
    "strconv"
)

func main() {
    data, err := ioutil.ReadFile("input.txt")
    if err != nil {
        fmt.Println("Error opening file")
        return
    }
    str := s.Trim(s.Replace(string(data), "\r", "", -1), "\n")
    list := s.Split(str, "\n")
    last := 0
    count := 0
    for i, line := range list {
        val, err := strconv.Atoi(line)
        if err != nil {
            fmt.Println("Error converting string to number: ", line)
        }
        if (i > 0 && last < val) {count += 1 }
        last = val
    }
    fmt.Println(count)
}

File: 2021/9/task1.py
lines = [x.rstrip() for x in open("input.txt")]
map = []
for line in lines:
    if line != "": map.append([int(x) for x in line])
H = len(map)
W = len(map[0])
count = 0
for y in range(H):
    for x in range(W):
        cur = map[y][x]
        if x > 0   and map[y][x-1]<=cur: continue
        if x < W-1 and map[y][x+1]<=cur: continue
        if y > 0   and map[y-1][x]<=cur: continue
        if y < H-1 and map[y+1][x]<=cur: continue
        count += 1 + int(cur)
print(count)
